numpy/_core/src/multiarray/textreading/rows.c

```diff
@@ -480,6 +480,14 @@ read_rows(stream *s,
         ((PyArrayObject_fields *)data_array)->dimensions[0] = row_count;
     }
 
+    /*
+     * If row_size is too big, F_CONTIGUOUS is always set
+     * as array was created for only one row of data.
+     * Clear the flag if more than one row was read.
+     */
+    if (row_count > 1) {
+        PyArray_CLEARFLAGS(data_array, NPY_ARRAY_F_CONTIGUOUS);
+    }
```

Comment:
I should have mentioned the other failure mode, since it is a bit hard to achieve:
```
assert np.loadtxt(["1 2", ""], ndmin=2).flags.f_contiguous
```
should succeed, but doesn't.  Thus to be perfectly right, we need to update the flags always, or add an else branch which sets it.

Maybe explain a bit more specific that depending on `row_size` the original array may or may not have been flagged as F_CONTIGUOUS, but that is wrong since the number of rows may have changed.  It is F_CONTIGUOUS exactly when `row_count <= 1`.

Reply:
Oops, sorry I didn't take it into consideration. Besides, it seems that even `np.loadtxt(["1 2"], ndmin=2).flags.f_contiguous` returns False. In this case, `row_count` still goes 2. I've added this sample into the test set.